packages/backend-common/src/database/connectors/sqlite3.ts

```diff
@@ -159,8 +162,250 @@ export function parseSqliteConnectionString(
+   *          `overridden`
+   */
+  private getClientType(pluginId: string): {
+    client: string;
+    overridden: boolean;
+  } {
+    const pluginClient = this.config.getOptionalString(
+      `${pluginPath(pluginId)}.client`,
+    );
+
+    const baseClient = this.config.getString('client');
+    const client = pluginClient ?? baseClient;
+    return {
+      client,
+      overridden: client !== baseClient,
+    };
+  }
+
+  private getRoleConfig(pluginId: string): string | undefined {
+    return (
+      this.config.getOptionalString(`${pluginPath(pluginId)}.role`) ??
+      this.config.getOptionalString('role')
+    );
+  }
+
+  /**
+   * Provides the knexConfig which should be used for a given plugin.
+   *
+   * @param pluginId - Plugin to get the knexConfig for
+   * @returns The merged knexConfig value or undefined if it isn't specified
+   */
+  private getAdditionalKnexConfig(pluginId: string): JsonObject | undefined {
+    const pluginConfig = this.config
+      .getOptionalConfig(`${pluginPath(pluginId)}.knexConfig`)
+      ?.get<JsonObject>();
+
+    const baseConfig = this.config
+      .getOptionalConfig('knexConfig')
+      ?.get<JsonObject>();
+
+    return merge(baseConfig, pluginConfig);
+  }
+
+  private getPluginDivisionModeConfig(): string {
+    return this.config.getOptionalString('pluginDivisionMode') ?? 'database';
+  }
+
+  /**
+   * Provides a Knex connection plugin config by combining base and plugin
+   * config.
+   *
+   * This method provides a baseConfig for a plugin database connector. If the
+   * client type has not been overridden, the global connection config will be
+   * included with plugin specific config as the base. Values from the plugin
+   * connection take precedence over the base. Base database name is omitted for
+   * all supported databases excluding SQLite unless `pluginDivisionMode` is set
+   * to `schema`.
+   */
+  private getConnectionConfig(pluginId: string): Knex.StaticConnectionConfig {
+    const { client, overridden } = this.getClientType(pluginId);
+
+    let baseConnection = normalizeConnection(
+      this.config.get('connection'),
+      this.config.getString('client'),
+    );
+
+    if (
+      client.includes('sqlite3') &&
+      'filename' in baseConnection &&
+      baseConnection.filename !== ':memory:'
+    ) {
+      throw new Error(
+        '`connection.filename` is not supported for the base sqlite connection. Prefer `connection.directory` or provide a filename for the plugin connection instead.',
+      );
+    }
+
+    // Databases cannot be shared unless the `pluginDivisionMode` is set to `schema`. The
+    // `database` property from the base connection is omitted unless `pluginDivisionMode`
+    // is set to `schema`. SQLite3's `filename` property is an exception as this is used as a
+    // directory elsewhere so we preserve `filename`.
+    if (this.getPluginDivisionModeConfig() !== 'schema') {
```

Comment:
hmm, is this check needed for anything but postgres?

Reply:
again, copy-pasted, needs cleanup later (this is the way it always was - a ton of redundant checks were made that only applied to some engines)